### PR TITLE
Fix Intel Auto-Detection and Profile Selection Menu

### LINKED ISSUE #2608: System profile prompting has inconsistent numbering
**What is the current experience of ilab you've found issue with?**
We have several menus where `[0]` is the "default" option but users can't actually see what it is and get misleading or confusing behavior if they select it

Example below - I hit enter without picking a number, and ilab assumed I picked NVIDIA.

```bash
Please choose a system profile.
 Profiles set hardware-specific defaults for all commands and sections of the configuration.
First, please select the hardware vendor your system falls into
[1] AMD
[2] APPLE
[3] INTEL
[4] NVIDIA
Enter the number of your choice [0]: 
[1] NVIDIA A100 X2
[2] NVIDIA A100 X4
[3] NVIDIA A100 X8
[4] NVIDIA H100 X2
[5] NVIDIA H100 X4
[6] NVIDIA H100 X8
[7] NVIDIA L4 X8
[8] NVIDIA L40S X4
[9] NVIDIA L40S X8
Enter the number of your choice [hit enter for hardware defaults] [0]: 
No profile selected - any hardware acceleration for training must be configured manually.
```

**How do you think the described experience should be?**
The first menu should list `[0] None of the below` and default to exiting early with no profile selected with the same message we give for the second menu, i.e. `No profile selected - any hardware acceleration for training must be configured manually.`

The second menu should list `[0] Hardware defaults (no profile)`

**What is the benefit of the change you are proposing?**
Much less confusing experience for the user

### PR DESCRIPTION
**Pull request by cdoern**
Intel returns Intel(r) for their CPUs, so we need to do a more general check than strict matching `intel`.

Also, add a [0] option for profiles that just sets the pydantic defaults. It is unclear who would get into this scenario with the auto-detection working
but if a user is running on such a system, the pydantic defaults might just be what they're looking f

Screenshot from 2024-11-11 21-19-20

resolves #2608 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

**Comment by nathan-weinberg** [comment]
cc @JustinXHale

**Comment by JustinXHale** [comment]
@cdoern with the auto detection will it bypass the [Choose system hardware] question all together? Or will users still be prompted to select?

**Comment by nathan-weinberg** [comment]
@cdoern can you post the output of the menu now similar to what I did with the original issue so we can easily compare?

**Comment by cdoern** [comment]
@nathan-weinberg is the unit test runner an x86 mac?

**Comment by cdoern** [comment]
if so, testing this is going to be realllllly tricky

**Comment by nathan-weinberg** [comment]
I don't know, but that should be a pretty simple thing to look up

**Comment by cdoern** [comment]
@nathan-weinberg @JustinXHale giving up on the no `[0]` front. Having no default:

1. broke our unit tests
2. will break anyone who has automation build around this probably

making a `[0]` which sets the defaults we store in code and applies no profile

I also fixed intel auto detect. 

Find a photo of the output attached to the PR description. thanks

File: src/instructlab/config/init.py [hunk 1]
```diff
@@ -213,7 +221,7 @@ def prompt_user_to_choose_profile(key, arch_family_processors) -> Config | None:
         cfg = read_config(file)
     elif system_profile_selection == 0:
         click.echo(
-            "No profile selected - any hardware acceleration for training must be configured manually."
+            "No profile selected - any hardware acceleration must be configured manually."
```
Comment (nathan-weinberg) [inline src/instructlab/config/init.py:224]: Is it accurate to just say that hardware acceleration must be configured manually?

I would suggest maybe something like 
`No profile selected - ilab will use generic code defaults - these may not be optimized for your system.`

Comment (cdoern) [reply src/instructlab/config/init.py:224]: that seems like kind of a mouthful, i feel like folks don't care that its the code defaults, but they do care that no acceleration was directly configured in their cfg (gpus: 0)

Comment (nathan-weinberg) [reply src/instructlab/config/init.py:224]: But the configuration being done extends to a lot more than hardware acceleration - I think that's worth calling out, as well as _where_ the values that are being used are coming from (if I don't have a profile, where is my config coming from?)

**Review by nathan-weinberg — APPROVED** [review]

**Pull request merged**